### PR TITLE
Create new 'flx' package. Just a copy for now.

### PR DESCRIPTION
**Pull request by mpcomplete**
This will be shared by sky/updater and flutter_tools. Currently
bundle.dart and signing.dart are unmodified copies of the files from
updater and flutter_tools, respectively. This is so future changes are
easier to see.

### CODE REVIEW

**Comment by mpcomplete** [comment]
I have another CL that will be based on this. Splitting it into 2 to make it easier to review.

File: sky/packages/flx/pubspec.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+name: updater
```
Comment (abarth) [inline sky/packages/flx/pubspec.yaml]: flx

File: sky/packages/flx/pubspec.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+version: 0.0.1
+author: Flutter Authors <flutter-dev@googlegroups.com>
+description: The autoupdater for flutter
```
Comment (abarth) [inline sky/packages/flx/pubspec.yaml]: Update description.

File: sky/packages/flx/pubspec.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+homepage: http://flutter.io
+dependencies:
+  mojo: '>=0.2.0 <0.3.0'
+  flutter: ">=0.0.3 <0.1.0"
```
Comment (abarth) [inline sky/packages/flx/pubspec.yaml]: This package doesn't depend on mojo or flutter

File: sky/packages/flx/pubspec.yaml [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  sky_services: any
+  path: any
+  yaml: any
+  cipher: any
+  asn1lib: any
```
Comment (abarth) [inline sky/packages/flx/pubspec.yaml]: Please fill in some version numbers here.

File: sky/packages/flx/pubspec.yaml [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+dependency_overrides:
+  flutter:
+    path: ../sky
```
Comment (abarth) [inline sky/packages/flx/pubspec.yaml]: You shouldn't need dependency_overrides.

**Comment by abarth** [comment]
Putting this package here creates an odd dependency cycle between the tools repo and the engine repo.  Should we just put it into its own repo?

**Comment by abarth** [comment]
I've created the https://github.com/flutter/flx repo for this code.  Please follow https://github.com/flutter/newton as a template.  This will also let you easily write unit tests using the normal Dart support in Travis.

**Comment by mpcomplete** [comment]
This package is heavily tied to flutter, and the updater especially. I don't see the value in having a separate repo, and it seems like it will make it more difficult to make changes, since every change will require updates to at least 2 repos.

The dependency cycle doesn't seem like an issue. This will be pushed to pub.dartlang.org, right? So flutter_tools will just use the version there, not the one from the flutter repo.

**Comment by abarth** [comment]
I feel like this discussion would be easier over VC.  Do you have a couple minutes to talk this over?

**Comment by abarth** [comment]
LGTM w/ some more sensible dependencies in the pubspec.yaml.

**Pull request merged**